.github/workflows/checks.yml

```diff
@@ -0,0 +1,56 @@
+name: Pull Request Checks
+
+on:
+  pull_request:
+    branches: ['main']
```

Comment:
No, the checks should run _both_ on pull requests against `main` AND on pushes to it.